### PR TITLE
feat: use redis to block requests

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes # (issue)
https://threads.com/thread/34565519413/34565519411?s=fSCVJ55JMMT6nAaL4VcyZX

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/api/book/event.ts [hunk 1]
```diff
@@ -17,8 +25,31 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
...
   /* To mimic API behavior and comply with types */
   req.userId = session?.user?.id || -1;
-  const booking = await handleNewBooking(req);
-  return booking;
+
+  const requestBody = JSON.stringify({ ...req.body, userIp });
+  const idempotencyKey = generateHash(requestBody);
+
+  try {
+    const checkIfKeyExist = await redis.exists(idempotencyKey);
```
Comment (keithwillcode) [inline apps/web/pages/api/book/event.ts:34]:
```suggestion
    const keyExists = await redis.exists(idempotencyKey);
```

File: apps/web/pages/api/book/event.ts [hunk 2]
```diff
@@ -17,8 +25,31 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
...
+    if (checkIfKeyExist) {
```
Comment (keithwillcode) [inline apps/web/pages/api/book/event.ts:35]:
```suggestion
    if (keyExists) {
```

File: apps/web/pages/api/book/event.ts [hunk 3]
```diff
@@ -17,8 +25,31 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
...
+    const booking = await handleNewBooking(req);
+    await redis.del(idempotencyKey);
+
+    return booking;
+  } catch (err: unknown) {
+    // Extract error information
+    const responseData = { message: err?.message ?? "Internal Server Error" };
+    const statusCode = err?.statusCode ?? 500;
+
+    await redis.del(idempotencyKey);
```
Comment (keithwillcode) [inline apps/web/pages/api/book/event.ts:50]: Nit: can just push the redis.del in a finally block so we don’t have this line twice

**Comment by Udit-takkar** [comment]
Fixed by Alex

**Pull request closed without merging**